pyproject.toml

```diff
@@ -54,7 +54,7 @@ duckb = [
     "duckdb>=0.10.3,<2 ; python_version >= '3.10' and python_version < '4.0'"
 ]
 datafusion = [
-    "datafusion>=34.0.0,<41; python_version >= '3.10' and python_version < '4.0'"
+    "datafusion>=34.0.0,<43; python_version >= '3.10' and python_version < '4.0'"
```

Comment:
why do we have a constraint on the datafusion version?

Reply:
I think is better to have it this way, having an open range means that the user may install a version of datafusion we haven't yet tested

Reply:
Ibis has a wider range than us
`datafusion = { version = ">=0.6,<41", optional = true }`
https://github.com/ibis-project/ibis/blob/9.5.0/pyproject.toml#L62
Can you point me to the version specific functionality we have?